src/ccl_background.c

```diff
@@ -451,39 +448,41 @@ void ccl_cosmology_compute_distances(ccl_cosmology * cosmo, int *status)
     gsl_spline_free(chi);
     *status=CCL_ERROR_MEMORY;
     ccl_cosmology_set_status_message(cosmo, "ccl_background.c: ccl_cosmology_compute_distances(): ran out of memory\n");
-    return; 
+    return;
   }
 
   a[0]=a0; a[na-1]=af;
   const gsl_root_fdfsolver_type *T=gsl_root_fdfsolver_newton;
   gsl_root_fdfsolver *s=gsl_root_fdfsolver_alloc(T);
   for(int i=1;i<na-1;i++) {
-    chistatus|=a_of_chi(y[i],cosmo, status, &a0,s);
+    // we are using the previous value as a guess here to help the root finder
+    // as long as we use small steps in a this should be fine
+    a_of_chi(y[i],cosmo, status, &a0,s);
     a[i]=a0;
   }
-  
+
   gsl_root_fdfsolver_free(s);
-  if(chistatus) {
+  if(splstatus) {
```

Comment:
Is this really splstatus?

Reply:
I changed the name of the variable and the code works. I don't follow the question though.

Reply:
So you already check for splstatus in 418. Then in line 460 there is "status" from the a_of_chi(), but this doesn't get checked. Instead, you check for splstatus again. Was this the intention?

Reply:
Ack yes. It looks like this bug was in the code already. I changed the name of the parameter only. I will fix it.